### PR TITLE
 Website: Combine the fetch logic in open-source page

### PR DESCRIPTION
**Pull request by activus-d**
This PR combines the fetch logic in [open-source](https://github.com/activus-d/socialincome-public/tree/main/website/src/app/%5Blang%5D/%5Bregion%5D/(website)/open-source) page for a more concise code.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for fetching data from GitHub repositories, streamlining data retrieval across various components.

- **Bug Fixes**
	- Simplified error handling across multiple functions, ensuring more consistent management of API responses.

- **Refactor**
	- Replaced direct fetch calls with a centralized data fetching utility, improving code modularity and maintainability across components.
	- Updated the structure of contributor data and removed redundant header management.

- **Documentation**
	- Updated function signatures to reflect changes in implementation while maintaining existing interfaces.

### CODE REVIEW

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.ts [hunk 1]
```diff
@@ -23,50 +25,12 @@ export async function getCommits() {
...
-		} else if (status === 404) {
-			throw new Error(`GitHub repository ${owner}/${repo} not found.`);
-		} else {
-			throw new Error(`Failed to fetch recent commits from GitHub: ${status} - ${errorDetails}`);
-		}
-	}
-
-	const recentCommits: GitHubCommit[] = await res.json();
+	const recentCommitsRes = await fetchData(owner, repo, commitUrl);
+	const recentCommits: GitHubCommit[] = await recentCommitsRes.json();
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.ts:29]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.ts [hunk 2]
```diff
@@ -14,38 +16,14 @@ interface IssuesResponse {
...
-					'GitHub API rate limit exceeded. Please try again later or increase rate limit by authenticating.',
-				);
-			} else if (status === 404) {
-				throw new Error(`GitHub repository ${owner}/${repo} not found.`);
-			} else {
-				throw new Error(`Failed to fetch issues from GitHub: ${status} - ${errorDetails}`);
-			}
-		}
-
+		const res = await fetchData(owner, repo, url);
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.ts]: fixed

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.ts [hunk 3]
```diff
@@ -17,31 +19,10 @@ interface GitHubContributor {
...
-		} else {
-			throw new Error(`Failed to fetch contributors from GitHub: ${status} - ${errorDetails}`);
-		}
-	}
-
-	const contributors = await res.json();
+export async function getContributors() {
+	const url = `https://api.github.com/repos/${owner}/${repo}/stats/contributors`;
+	const contributorsRes = await fetchData(owner, repo, url);
+	const contributors = await contributorsRes.json();
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.ts:25]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/fetch-data.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		if (status === 403) {
+			throw new Error(
+				'GitHub API rate limit exceeded. Please try again later or increase rate limit by authenticating.',
+			);
+		} else if (status === 404) {
+			throw new Error(`GitHub repository ${owner}/${repo} not found.`);
+		} else {
+			throw new Error(`Failed to fetch recent commits from GitHub: ${status} - ${errorDetails}`);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/fetch-data.ts:26]: not relevant

**Review by mkue — APPROVED** [review]
Nice! Thanks for cleaning this up.

**Pull request merged**